### ISSUE TITLE
Understanding XGBClassifier Results from Force_plot

### ISSUE THREAD

**Issue by SaadAhmed96** [shap/shap#483]
So I have been working on understanding how this library works and interpretating results is somewhat confusing for me. You can look at the picture attached here. I first fitted a XGBClassifier with objective  = "multi:softprob" and as you can see from the force_plot that I get a logit value of -0.69 indicating that the likelihood of the occurrence of class [0] is less and for the class [1] I get a logit value of 1.69 indicating a higher chance of its occurrence.

![image](https://user-images.githubusercontent.com/30757857/54073683-f93ee400-42ab-11e9-903a-7e0927c7a458.png)

I am confused because as this is a binary classification problem I assumed that the logit value for the class[1] would be 0.69 instead of 1.69. If anyone can enlighten me about this it would be very helpful.
EDIT: Also when I  use XGBClassifier(objective ="binary:logistic") and calculate the shap_values and expected values I only get the shap_values for the second class and the expected value for that as well. So how am I supposed to plot the force_plot for the first class  for my classification task as I cant change the indexing of the first two arguments of the force_plot.

Thanks
Saad

**Comment by slundberg** [comment]
The softmax operator normalizes everything, so there is no need for things to be centered at 0. (note this observation doesn't really involve SHAP values since both those numbers are just the raw output from XGB)

**Comment by SaadAhmed96** [comment]
@slundberg thanks for the insight! One thing I noticed when I plotted force plot different examples in my test set is that when I sum the two classes output value is always equal to 1 which I can't understand. Is there any particular reason why their sum is always centered towards 1?

**Comment by SaadAhmed96** [comment]
![image](https://user-images.githubusercontent.com/30757857/54183196-5b266600-44c5-11e9-8747-836a675c11b6.png)
Also please look at the pic attached and let me know why I am only getting expected value for just one class and how am I supposed to know that this expected value is for the first class or the second. I know I am asking too many questions but kindly bear with me as I have recently started towards working on interpreting m.l models.


I also thought that I should be getting a list of size 2 with each element having size of (samples, features) but instead its just a single vector of size  (samples, features). 


Thanks

**Comment by digivi13** [comment]
I also have the same issue as stated above using xgboost binary classifier, I only get a single list that seems to correspond to the first class (classes are 0,1)

**Comment by slundberg** [comment]
@SaadAhmed96 (sorry for the slow reply) and @digivi13 this is an artifact of how XGBoost deals with binary classifiers. Since XGBoost only outputs a single vector of numbers (for the "1" class"), so also SHAP only explains a single output. The probability for one class (or log odds) is determined by the probability of the other class. So that means a positive contribution towards class 0 is just as easily interpreted as a negative contribution towards class 1.

**Comment by digivi13** [comment]
@slundberg  I kind of figured that may be the case. It was also good to find out that the output is in log odds by default for xgboost models. Using link="logit" helped me get force plots with the proper probabilities. If this could also be done for summary and dependance plots, that would be awesome.

**Comment by slundberg** [comment]
@digivi13 the reason we can use a link function for the force plot is because we have all the features together and so they add up to the actual output. In a dependence or summary plot we have taken just the impact of a feature all by itself, and this can't be transformed after-the-fact by just a scale change. If you want probabilities for everything then I would suggest using the `model_output="probability"` option of TreeExplainer.

**Comment by digivi13** [comment]
That makes sense, thanks for the support on this library! I think this could be closed now based on your responses.
